core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,243 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.error
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.os.Process
+import android.view.View
+import android.widget.Button
+import android.widget.CheckBox
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import butterknife.BindView
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R2
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.dao.NewBookDao
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.getCurrentLocale
+import org.kiwix.kiwixmobile.core.utils.files.FileLogger
+import org.kiwix.kiwixmobile.zim_manager.MountPointProducer
+import java.io.PrintWriter
+import java.io.StringWriter
+import javax.inject.Inject
+import kotlin.system.exitProcess
+
+open class ErrorActivity : BaseActivity() {
+  @Inject
+  var bookDao: NewBookDao? = null
+
+  @Inject
+  var zimReaderContainer: ZimReaderContainer? = null
+
+  @Inject
+  var mountPointProducer: MountPointProducer? = null
+
+  @Inject
+  var fileLogger: FileLogger? = null
+
+  @BindView(R2.id.reportButton)
+  var reportButton: Button? = null
```

Comment:
Views do not need to be injected in Kotlin thanks to Kotlin Android Extensions, https://antonioleiva.com/kotlin-android-extensions/

This means that you can simply reference them with e.x. `reportButton.setOnClickListener...` without having to keep them as properties.

Reply:
Thanks for your help @Frans-Lukas